src/lib/channel.ts

```diff
@@ -42,3 +43,79 @@ export const channelIdToPathString = (
   const simpleChannelPath = channelIdToSimpleChannelPath(channelId, channelMap)
   return simpleChannelPath.map(c => c.name).join('/')
 }
+
+export type checkResult = 'none' | 'match' | 'perfect'
+
+export interface matchResult<T extends ChannelLike> {
+  perfectMatched: T[]
+  matched: T[]
+}
+
+const checkMatchChannel = (
+  channel: ChannelLike,
+  query: string
+): checkResult => {
+  if (channel.name === query) return 'perfect'
+  if (channel.name.includes(query)) return 'match'
+  return 'none'
+}
+
+/**
+ * 連続するチャンネルに対し、連続する条件を満たすようなチャンネルを得る関数
+ * @param channelMap チャンネルの id とチャンネル情報を対応付ける map
+ * @param querys 連続するクエリ
+ * @param targetChannelMap 対象のチャンネルの map
+ * @returns 条件を満たすようなチャンネルの配列
+ */
+export const channelDeepMatching = <T extends ChannelLike>(
+  channelMap: ReadonlyMap<ChannelId, T>,
+  querys: [string, ...string[]],
```

Comment:
```suggestion
  querys: readonly [string, ...string[]],
```
にしてもよさそうです (`channelRecursiveDeepMatching`のほうも)

Reply:
両方 `readonly` つけました :person_bowing: